src/py/flwr/client/node_state.py

```diff
@@ -26,13 +29,23 @@ class NodeState:
     def __init__(self, partition_id: Optional[int]) -> None:
         self._meta: Dict[str, Any] = {}  # holds metadata about the node
         self.run_contexts: Dict[int, Context] = {}
+        self._initial_run_config: Dict[str, str] = {}
```

Comment:
We need to store the initial `run_config` for each `run_id`:
```suggestion
        self.run_contexts: Dict[int, Context, Dict[str, str]] = {}
```